examples/slurm/README.md

```diff
@@ -0,0 +1,215 @@
+
+    Use ```sbatch --nodelist=idle01 start_head_node.sh``` to submit the script to certain idle node (here we assume it is ```idle01```).
+
+4. **Expected output**
+
+   In ```sllm_head.out```, you will see the following output:
+   ```shell
+   Local node IP: xxx.xxx.xx.xx
+   --------------------
+   Ray runtime started.
+   --------------------
+   ```
+   Remember the IP address, denoted ```<HEAD_NODE_IP>```, you will need it in following steps.
+
+## Step 2: Start the Worker Node
+1. **Activate the ```sllm-worker``` environment and start the worker node.**
+
+   Here is the example script, named```start_worker_node.sh```.
+   ```shell
+   #!/bin/sh
+   #SBATCH --partition=your_partition
+   #SBATCH --job-name=sllm-worker
+   #SBATCH --output=sllm_worker.out
+   #SBATCH --error=sllm_worker.err
+   #SBATCH --gres=gpu:2                   # Request 2 GPUs
+   #SBATCH --cpus-per-task=4              # Request 4 CPU cores
+   #SBATCH --mem=16G                      # Request 16GB of RAM
+
+   cd /path/to/ServerlessLLM
+
+   conda activate sllm-worker
+
+   HEAD_NODE_IP=<HEAD_NODE_IP>
+
+   ray start --address=$HEAD_NODE_IP:6379 --num-cpus=4 --num-gpus=2 \
+   --resources='{"worker_node": 1, "worker_id_0": 1}' --block
+   ```
+   - Read the HPC's documentation to find out which partition you can use. Replace ```your_partition``` in the script with that partition name.
+   - Replace ```/path/to/ServerlessLLM``` with the path to the ServerlessLLM installation directory.
+   - Replace ```<HEAD_NODE_IP>``` with the IP address of the head node.
+2. **Submit the script on the other node**
+
+    Use ```sbatch --nodelist=idle02 start_worker_node.sh``` to submit the script to certain idle node (here we assume it is ```idle02```). In addition, We recommend that you place the head and worker on different nodes so that the Store and Serve can start smoothly later, rather than queuing up for resource allocation.
+3. **Expected output**
+
+   In ```sllm_worker.out```, you will see the following output:
+   ```shell
+   Local node IP: xxx.xxx.xx.xx
+   --------------------
+   Ray runtime started.
+   --------------------
+   ```
+
+## Step 3: Start the Store on the Worker Node
+1. **Activate the ```sllm-worker``` environment and start the store.**
+
+   Here is the example script, named```start_store.sh```.
+   ```shell
+   #!/bin/sh
+   #SBATCH --partition=your_partition
+   #SBATCH --gres=gpu:1                   # Request 1 GPU
+   #SBATCH --cpus-per-task=4              # Request 4 CPU cores
+   #SBATCH --mem=16G                      # Request 16GB of RAM
+   #SBATCH --time=01:00:00                # Run time (hh:mm:ss)
+   #SBATCH --output=store.log             # Standard output and error log
+
+   cd /path/to/ServerlessLLM
+
+   conda activate sllm-worker
+
+   export CUDA_HOME=/opt/cuda-12.5.0 # replace with your CUDA path
+   export PATH=$CUDA_HOME/bin:$PATH
+   export LD_LIBRARY_PATH=$CUDA_HOME/lib64:$LD_LIBRARY_PATH
+
+   sllm-store-server
```

Comment:
`sllm-store-server` should run on the same GPU with worker node. This is because `sllm-store-server` will load the model weights into GPU and use shared GPU memory to pass the pointer to the client. If you use another script with `--gpres=gpu:1`, they will be in different GPUS setting by different `CUDA_VISIBLE_DEVICES`. Thus, they cannot pass the model weights. I suggest we can write the start the store server on work node in the same script with start worker node.

Reply:
No problem, thanks a lot for suggestions!